determined-ai/determined
# fix: Poll roles, remove canGetPermissions permission

## Description

I'm trying to resolve ongoing issues with the Workspace Members page. Currently on an RBAC/EE server, the row has the current role's number ID and no selectable roles.

I am making two changes here:
- Using `useFetchKnownRoles` to populate the roles dropdown in Members page.
- The above change also helps fix the Add Member modal. Any role can be assigned to a workspace member.
- Change/remove workspace member role is now functioning from API change
- ~Removing `canGetPermissions` since it was an alias for the Assign Roles permission, and we should be able to view permissions / members of the workspace even if we can't assign roles.~

## Test Plan

- Use a determined-ee environment to fully test API and client
- Log in as admin
- Create a workspace. The table in the members tab should include admin ~(as admin, you are not shown)~
- In the members tab, click the Add Members button, and add 'determined' to the workspace with a Viewer role. 
- Confirm that determined is shown in the table now and after reloading.
- Use the determined row's dropdown to change role to Editor. Confirm that the new role is shown now and after reloading the page.
- Log in as determined, and confirm that you can view the workspace (you will not see members listed).
- Log in as admin again. In the members table, on the right end of the 'determined' row, you should have a three-dots menu. Click that button to open a dropdown and click 'Remove'. In this modal, confirm removing this role from 'determined'.

## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
